webapp/src/Cache.ts

```diff
@@ -17,10 +18,13 @@ export class Cache {
       await ServerConfig.getProjectConfig(projectName);
     const repoPath = serverProjectConfig.repoPath;
     const lyraConfig = await LyraConfig.readFromDir(repoPath);
-    await Cache.gitPullIfNeeded(repoPath, lyraConfig.baseBranch);
     const lyraProjectConfig = lyraConfig.getProjectConfigByPath(
       serverProjectConfig.projectPath,
     );
+    if (!lyraProjectConfig.isLanguageSupported(lang)) {
+      throw new LanguageNotSupported(lang, projectName);
+    }
+    await Cache.gitPullIfNeeded(repoPath, lyraConfig.baseBranch);
```

Comment:
I think I understand why we move gitPullIfNeeded: to early-out on invalid data.

However, LyraConfig is stored in the repository so it might need re-reading after pull. Consider for example if a language is added, then Lyra will continue to reject these requests until restarted, or possible until git pull has been performed by the pull-request route. I expect this is a much bigger problem as there is plenty of state in Lyra already that might actually need to be invalidated by pull.

Right now, I fear it is Only safe to update the target project via Lyra and that any update from outside Lyra needs to be followed up by an immediate restart of Lyra.

Let's open another issue to investigate that problem domain and not let it block this pull request.

Reply:
But before it was also loaded after reading from repo, and it has to be exec after LyraConfig.readFromDir because we need the value of the base branch name (which is mostly "main")

How it was and how it become is not the best setup since we need to know the name of the branch and this info could by it self stored in yet new or unknown branch name !! 

Agree it need another issue, I suggest to move baseBranch setting to ServerConfig and delete it from LyraConfig

Reply:
This is being moved up again in https://github.com/zetkin/lyra/pull/54/files#diff-89849ad497e92442823fdc23171341af1b9049a6ff0bba4f5bfa9159908b6adb

Reply:
I just reviewed 4ab77cb4b4284485235b5f51e7899faa2501d19c for things that might need to be invalidated after pull and I found just LyraConfig and Translations.

Translations is already covered by https://github.com/zetkin/lyra/issues/25
For LyraConfig, I just created https://github.com/zetkin/lyra/issues/84